web-common/src/features/connectors/explorer/TableMenuItems.svelte

```diff
@@ -70,16 +71,40 @@
       console.error(err);
     }
   }
+
+  async function handleCreateModelFromTable() {
+    if (isModelingSupported) {
+      await handleCreateModel(() =>
+        createSqlModelFromTable(
+          queryClient,
+          connector,
+          database,
+          databaseSchema,
+          table,
+        ),
+      );
+    } else if (showGenerateModel) {
+      await handleCreateModel(() =>
+        createYamlModelFromTable(
+          queryClient,
+          connector,
+          database,
+          databaseSchema,
+          table,
+        ),
+      );
+    }
+  }
 </script>
 
-{#if isModelingSupported}
-  <NavigationMenuItem on:click={handleCreateModel}>
+{#if isModelingSupported || showGenerateModel}
+  <NavigationMenuItem on:click={handleCreateModelFromTable}>
     <Model slot="icon" />
-    Create new model
+    Create model
   </NavigationMenuItem>
 {/if}
 
-{#if showGenerateMetricsAndDashboard}
+{#if showGenerateMetricsAndDashboard && isModelingSupported}
```

Comment:
This doesn't look like it'll work for Druid & Pinot & self-hosted ClickHouse. For all these cases, we don't support modeling, but we do want the user to be able to create metrics views on top of their OLAP's tables.

Reply:
https://github.com/rilldata/rill/pull/7716/commits/c270fb11719aa75899ff5da98194c0342bfe93c7